operators/pkg/controller/elasticsearch/sset/list.go

```diff
@@ -81,6 +83,22 @@ func (l StatefulSetList) GetActualPods(c k8s.Client) ([]corev1.Pod, error) {
 	return allPods, nil
 }
 
+// PodReconciliationDone returns true if actual existing pods match what is specified in the StatefulSetList.
+// It may return false if there are pods in the process of being created (but not created yet)
+// or terminated (but not removed yet).
+func (l StatefulSetList) PodReconciliationDone(c k8s.Client, es v1alpha1.Elasticsearch) (bool, error) {
+	// pods we expect to be there based on StatefulSets spec
+	expectedPods := l.PodNames()
+	// pods that are there for this cluster
+	actualRawPods, err := GetActualPodsForCluster(c, es)
+	if err != nil {
+		return false, err
+	}
+	actualPods := k8s.PodNames(actualRawPods)
+	// check if they match
+	return len(expectedPods) == len(actualPods) && stringsutil.StringsInSlice(expectedPods, actualPods), nil
```

Comment:
Would it make sense to have a debug log statement to find out which pods are keeping us from making progress?